.github/workflows/build.yml

```
@@ -0,0 +1,51 @@
+name: build
+
+on:
+  release:
+    types: ["published"]
+  pull_request:
+    branches: ["main"]
+
+env:
+  GOLANG_VERSION: "1.15"
+  BINARY_NAME: "os-agent-debian-amd64.bin"
+
+jobs:
+  build_debian:
+    name: Build amd64 debian
+    runs-on: ubuntu-latest
+    container: buildpack-deps:buster
+    steps:
+    - name: Checkout the repository
+      uses: actions/checkout@v2
+
+    - name: Setup go ${{ env.GOLANG_VERSION }}
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GOLANG_VERSION }}
+
+    - name: Build binary
+      run: |
+        version=$(echo "${{ github.ref }}" | awk -F"/" '{print $NF}')
+        go build -ldflags="-s -w -X main.version=${version}" -o "${{ env.BINARY_NAME }}"
+        du -hs *
```

```suggestion
        du -hs os-agent-debian-amd64.bin
```
Do we care about other files/directories?